# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rustc/hotfix-i686-no-sse2.patch
# Copyright (C) 2025 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- rustc-1.93.0-src/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs.vanilla	2026-01-22 17:23:30.028391422 +0100
+++ rustc-1.93.0-src/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs	2026-01-22 17:25:33.705993801 +0100
@@ -1,11 +1,11 @@
 use crate::spec::{
-    Arch, Cc, LinkerFlavor, Lld, RustcAbi, SanitizerSet, StackProbeType, Target, TargetMetadata,
+    Arch, Cc, LinkerFlavor, Lld, SanitizerSet, StackProbeType, Target, TargetMetadata,
     base,
 };
 
 pub(crate) fn target() -> Target {
     let mut base = base::linux_gnu::opts();
-    base.rustc_abi = Some(RustcAbi::X86Sse2);
+    base.rustc_abi = None;
     // Dear distribution packager, if you are changing the base CPU model with the goal of removing
     // the SSE2 requirement, make sure to also set the `rustc_abi` to `None` above or else the compiler
     // will complain that the chosen ABI cannot be realized with the given CPU features.
@@ -16,7 +16,7 @@
     // can lead to memory safety violations if floating-point values are used e.g. to access an
     // array. If users run into such issues and report bugs upstream and then it turns out that the
     // bugs are caused by distribution patches, that leads to confusion and frustration.
-    base.cpu = "pentium4".into();
+    base.cpu = "i686".into();
     base.max_atomic_width = Some(64);
     base.supported_sanitizers = SanitizerSet::ADDRESS;
     base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-m32"]);
